test/hack/resource/pkg/resourcetypes/securitygroup.go

```diff
@@ -43,96 +43,69 @@ func (sg *SecurityGroup) Global() bool {
 }
 
 func (sg *SecurityGroup) GetExpired(ctx context.Context, expirationTime time.Time, excludedClusters []string) (ids []string, err error) {
-	var nextToken *string
-	for {
-		out, err := sg.ec2Client.DescribeSecurityGroups(ctx, &ec2.DescribeSecurityGroupsInput{
-			Filters: []ec2types.Filter{
-				{
-					Name:   lo.ToPtr("group-name"),
-					Values: []string{"security-group-drift"},
-				},
+	sgs, err := sg.getAllSecurityGroups(ctx, &ec2.DescribeSecurityGroupsInput{
```

Comment:
Have you considered just passing in `ec2types.Filter` instead of the whone `DescribeSecurityGroupsInput`?

Reply:
I hadn't because all the `Paginator` functions take in their respective `Input`, and I thought it was clearer to pass `Input` in instead of the struct field `Filter` into the `getAll<Resource>` functions. 

It also felt more general, if down the road, we wanted to `getAll<Resource>` based on other fields of the struct (like `MaxResults`), but I'm not sure if that necessary to consider in these cases.

Do you think it would be better to pass in an optional `filter` across all the resources instead? Something like: 
```
func (sg *SecurityGroup) getAllSecurityGroups(ctx context.Context, filters ...ec2types.Filter) (sgs []ec2types.SecurityGroup, err error) {
    params := &ec2.DescribeSecurityGroupsInput{
        Filters: filters,
    }
```

Reply:
That works with me